Extensions/MUC-Sub/XMPPMUCSub.m

```diff
@@ -0,0 +1,699 @@
+                // Must have been another request with the same id?
+                // Can this actually happen? Still, safeguard for my conscience.
+                return;
+            }
+            
+            [multicastDelegate xmppMUCSub:self didReceiveSubscribersIn:rooms to:iq.from];
+        }
+        else {
+            [multicastDelegate xmppMUCSubDidFailToReceiveSubscribersIn:self to:iq.from
+                                                                 error:[self errorFromIQ:iq]];
+        }
+    }};
+    
+    if (dispatch_get_specific(moduleQueueTag)) {
+        block();
+    }
+    else {
+        dispatch_async(moduleQueue, block);
+    }
+}
+
+
+- (void)handleSupportedByIQ:(XMPPIQ *)iq withInfo:(XMPPBasicTrackingInfo *)basicTrackingInfo
+{
+    // <iq from='coven@muc.shakespeare.example'
+    //       to='hag66@shakespeare.example/pda'
+    //     type='result'
+    //       id='ik3vs715'>
+    //   <query xmlns='http://jabber.org/protocol/disco#info'>
+    //     <identity category='conference'
+    //                   name='A Dark Cave'
+    //                   type='text' />
+    //     <feature var='http://jabber.org/protocol/muc' />
+    //     ...
+    //     <feature var='urn:xmpp:mucsub:0' />
+    //     ...
+    //   </query>
+    // </iq>
+    
+    dispatch_block_t block = ^{ @autoreleasepool {
+        if (iq.isResultIQ) {
+            NSXMLElement *query = [iq elementForName:@"query" xmlns:XMPPDiscoInfoNamespace];
+            for (NSXMLNode *child in query.children) {
+                if ([@"feature" isEqualToString:child.name] && NSXMLElementKind == child.kind) {
+                    NSString *var = [(NSXMLElement *)child attributeStringValueForName:@"var"];
+                    if ([XMPPMUCSubNamespace isEqualToString:var]) {
+                        [multicastDelegate xmppMUCSub:self serviceSupportedBy:iq.from];
+                        return;
+                    }
+                }
+            }
+            
+            [multicastDelegate xmppMUCSub:self serviceNotSupportedBy:iq.from];
+        }
+        else {
+            [multicastDelegate xmppMUCSub:self didFailToReceiveSupportedBy:iq.from
+                                                                 error:[self errorFromIQ:iq]];
+        }
+    }};
+    
+    if (dispatch_get_specific(moduleQueueTag)) {
+        block();
+    }
+    else {
+        dispatch_async(moduleQueue, block);
+    }
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+#pragma mark -
+#pragma mark Message + Presence Delegate
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+- (BOOL)xmppStream:(XMPPStream *)sender didReceiveIQ:(XMPPIQ *)iq
+{
+    NSString *type = iq.type;
+    
+    if ([type isEqualToString:@"result"] || [type isEqualToString:@"error"]) {
+        return [xmppIDTracker invokeForID:iq.elementID withObject:iq];
+    }
+    
```

Comment:
What does this do, delegate the decision whether this was an IQ response we waited for to `xmpppIDTracker`?

Reply:
Yes.